avm/ptn/deployment-script/import-image-to-acr/main.bicep

```
@@ -0,0 +1,255 @@
+param subnetId string = ''
+
+@description('Optional. Tags of the resource.')
+@metadata({
+  example: '''
+  {
+      "key1": "value1"
+      "key2": "value2"
+  }
+  '''
+})
+param tags object?
+
+// the following settings are only needed if the deployment script is using PrivateLink
+// they must only be added if they contain values. Conditional values for the properties does not work and ARM throws an error
+var storageSettings = !empty(storageAccountName)
+  ? {
+      storageAccountSettings: {
+        storageAccountName: storageAccountName
+      }
+    }
+  : {}
+var containerSettings = !empty(subnetId)
+  ? {
+      containerSettings: {
+        // an existing subnet is needed, if PrivateLink is going to be used
+        subnetIds: [
+          {
+            id: subnetId
+          }
+        ]
+        containerGroupName: '${resourceGroup().name}-infrastructure'
+      }
+    }
+  : {}
+
+// ============== //
+// Resources      //
+// ============== //
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.deploymentscript-importimagetoacr.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource acr 'Microsoft.ContainerRegistry/registries@2023-07-01' existing = {
+  name: acrName
+}
+
+resource newManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = if (!useExistingManagedIdentity) {
+  name: managedIdentityName
+  location: location
+  tags: tags
+}
+
+resource existingManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = if (useExistingManagedIdentity) {
+  name: managedIdentityName
+  scope: resourceGroup(existingManagedIdentitySubId, existingManagedIdentityResourceGroupName)
+}
+
+resource rbac 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = if (!empty(rbacRoleNeeded)) {
+  name: guid(acr.id, rbacRoleNeeded, useExistingManagedIdentity ? existingManagedIdentity.id : newManagedIdentity.id)
+  scope: acr
+  properties: {
+    roleDefinitionId: resourceId('Microsoft.Authorization/roleDefinitions', rbacRoleNeeded)
+    principalId: useExistingManagedIdentity
+      ? existingManagedIdentity.properties.principalId
+      : newManagedIdentity.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource importImage 'Microsoft.Resources/deploymentScripts@2023-08-01' = {
+  name: 'ACR-Import-${name}-${last(split(replace(image,':','-'),'/'))}'
```

would it maybe make sense to change this to 
```suggestion
  name: name ?? 'ACR-Import-${last(split(replace(image,':','-'),'/'))}'
```
assuming the `name` would be nullable? Who ever deploys this pattern may have a naming convention